crates/evm/src/system_calls/mod.rs

```diff
@@ -321,4 +318,11 @@ where
 
         eip7251::post_commit(result_and_state.result)
     }
+
+    /// Delegate to stored `OnStateHook`, noop if hook is `None`.
+    pub fn on_state(&mut self, state: &ResultAndState) {
+        if let Some(ref mut hook) = &mut self.hook {
+            hook.on_state(state);
+        }
+    }
```

Comment:
I think for now this is perfectly fine, I assume there's a better approach to this once we try to redesign the executor parts